### PR TITLE
test(core): Add unit tests for minimum-order-amount promotion condition

### PR DESCRIPTION
**Pull request by GabrielRoc**
## Description

Adds a direct unit test suite for the `minimum_order_amount` promotion condition (`min-order-amount-condition.ts`), which previously had **0% function coverage**.

The condition has two branches controlled by the `taxInclusive` boolean arg:

- `taxInclusive = false` → compares `order.subTotal` against `amount`
- `taxInclusive = true` → compares `order.subTotalWithTax` against `amount`

8 tests covering both branches via **boundary-value analysis** (threshold ±1, exact threshold) and confirming that the inactive field is correctly ignored in each mode.

All tests are database-free; `Order` instances are built in memory with explicit `subTotal` / `subTotalWithTax` values.

## Breaking changes

None.

## Checklist

- [x] I have set a clear title
- [x] My PR is small and contains a single feature
- [x] I have added or updated test cases

Relates to #4835

---
<a href="https://app.blacksmith.sh/vendurehq/codesmith/vendure/pr/4880"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1785127803&installation_id=128408429&pr_number=4880&repository=vendurehq%2Fvendure&return_to=https%3A%2F%2Fgithub.com%2Fvendurehq%2Fvendure%2Fpull%2F4880&signature=556eebd7e7ab615f289171c5682418b77a226c26fc83ec73b27e7449fc0c9392"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

**Comment by GabrielRoc** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by GabrielRoc** [comment]
recheck

File: packages/core/src/config/promotion/conditions/min-order-amount-condition.spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+function orderWith(values: { subTotal: number; subTotalWithTax: number }): Order {
+    return new Order({ lines: [], subTotal: values.subTotal, subTotalWithTax: values.subTotalWithTax });
+}
+
+function check(order: Order, args: ConfigArg[]) {
+    const ctx = createRequestContext({ pricesIncludeTax: false });
+    return minimumOrderAmount.check(ctx, order, args, undefined as any);
+}
```
[the remark this answers is not in the document]
Comment (GabrielRoc) [reply packages/core/src/config/promotion/conditions/min-order-amount-condition.spec.ts:23]: Dismissed. Passing `undefined as any` for the `promotion` argument is the established pattern across all promotion-condition specs in this PR series (including the template `buy-x-get-y-free-condition.spec.ts`). The `minimumOrderAmount.check()` implementation does not reference the `promotion` parameter at all, so the cast is safe and avoids constructing a heavyweight `Promotion` entity for no functional benefit.

**Review by grolmus — APPROVED** [review]
Excellent, well-structured coverage — boundary-value analysis on the thresholds, clean injector-based mocking (no DB), and assertions that the condition is invoked with the right args. Test-only, no production changes. Approving.
